.github/workflows/scanner-versioned-definitions-update.yaml

```diff
@@ -350,17 +367,28 @@ jobs:
       uses: google-github-actions/setup-gcloud@v3
 
     - name: Upload to GCS
+      env:
+        USE_RC: ${{ github.event.inputs.use_rc }}
       run: |
         for dir in definitions_files/*; do
           if [ -d "$dir" ]; then
-            echo "Copy $dir"
-            gsutil -m cp -r "$dir" gs://definitions.stackrox.io/v4/vulnerability-bundles/
+            name=$(basename "$dir")
+            dest="vulnerability-bundles/$name"
+            if [ "$USE_RC" = "true" ]; then
+              dest="vulnerability-bundles/rc/$name"
+            fi
```

Comment:
Should this condition to not overwrite 'live' bundles also include a condition for `stream` or `reference` being set? Thinking whoever dispatches the job could set something like:

`stream` = `v2`
`reference` = `heads/my-testing-branch`

And then live `v2` would written w/ the users custom branch (IIUC)

Reply:
Yes, one could dispatch from a branch with the relevant files changed and the effect would be similar. In comparison, dispatching is more fragile, as the params are inputs, rather than modified files.

Allowing dispatching with `use_rc=False` is powerful, but easy to misuse.

One alternative is to replace `use_rc` with `not_safe_publish_to_customers`:

1. If false (the default) it will always add `-rc` to `stream` if `stream != ''`, and have the same effect as `use_rc=True` today.
2. If true, it will not add the `-rc` to `stream`, and have the same effect as `use_rc=False`.

With the above, the most common use case (`stream` = `v2` and `reference` = `X.Y.Z`) would default to publishing `v2-rc`. But, if needed, there is a flag to overwrite specific stream builds from a dispatch.

Reply:
Another alternative is replacing `use_rc` and `reference`, with `production_ref` and add `release_candidate_ref`:

| `release_candidate_ref` | `production_ref` | Outcome |
|-|-|-|
|X.Y.Z or git ref| not set | Same as using `use_rc=True`, publish to `vN-rc` |
| not set | X.Y.Z or git ref | Same as using `use_rc=False`, publish to `vN` (production bundle)|
| X.Y.Z etc | X.Y.Z etc | Fails / Abort |
| not set | not set | Fails / Abort |

We could fashion a `GET` to https://raw.githubusercontent.com/stackrox/stackrox/ACTUAL_GIT_REFERENCE/scanner/VULNERABILITY_VERSION to determine the stream version to publish, if <stream> is empty -- so to avoid further human mistakes.

Reply:
OK, after reviewing these options on my own I am leaning towards the second.

Pros:

1. Chances of backfire reduced significantly. Clear outcome based on the inputs you're setting/modifying. Don't touch `production_ref` unless you want to mess with production.
2. RC-behavior is decoupled from the input. As in, it abstracts it. Scripts calling these during pre-release don't care if this is "rc-based versioning" etc.
3. One more automation step if we perform the `GET` on the stream version for the release candidate (or production).

Cons:

1. Chances of backfire are not 0.